src/realm/table.cpp

```diff
@@ -4965,13 +4970,21 @@ void Table::update_from_parent(size_t old_baseline) noexcept
     if (!m_columns.is_attached())
         return; // Degenerate subtable
 
-    if (!m_columns.update_from_parent(old_baseline))
-        return;
-
-    // Update column accessors
-    for (auto& col : m_cols) {
-        if (col != nullptr) {
-            col->update_from_parent(old_baseline);
+    if (m_columns.update_from_parent(old_baseline)) {
+        // Update column accessors
+        for (auto& col : m_cols) {
+            if (col != nullptr) {
+                col->update_from_parent(old_baseline);
+            }
+        }
+    }
+    else if (spec_might_have_changed) {
```

Comment:
No, but doing the `update_from_parent` includes refreshing the subtable maps. The addition is to ensure that we update the subtable maps even if the columns are not touched.